snapcraft/internal/lifecycle.py

```diff
@@ -88,6 +97,15 @@ def init():
     with open(snapcraft_yaml_path, mode='w') as f:
         f.write(yaml)
 
+    git_cache = os.path.join('.git')
```

Comment:
The use of `os.path.join` is unnecessary, here. In fact, I think this entire variable is unnecessary, no?